utils/utils.go

```diff
@@ -41,4 +43,66 @@ func GetAbsPath(path string) string{
 		}
 	}
 	return result
-}
\ No newline at end of file
+}
+
+func GetCurrentServerVersion() (string, string) {
+	var version, product string
+	rhndefault := "/etc/rhn/rhn.conf"
+	webpath := "/usr/share/rhn/config-defaults/rhn_web.conf"
+	altpath := "/usr/share/rhn/config-defaults/rhn.conf"
+
+	files := []string{rhndefault, webpath, altpath}
+	property := []string{"product_name", "web.product_name"}
+	product = "SUSE Manager"
+	for _, path:= range files {
+		for _, search := range property {
+			p, err := ScannerFunc(path, search)
+			if err == nil {
+				product = p
+				break
+			} else if err != nil {
+				continue
+			}
+		}
+	}
+
+	if product != "SUSE Manager" {
+		product = "uyuni"
+		v, err := ScannerFunc(rhndefault, "web.version.uyuni")
+		if err != nil {
+			v, err = ScannerFunc(webpath, "web.version.uyuni")
+			if err != nil {
+				log.Fatal().Msgf("No version found for product %s", product)
+			}
+		}
+		version = v
+	} else if product == "SUSE Manager" {
+		v, err := ScannerFunc(rhndefault, "web.version")
+		if err != nil {
+			v, err = ScannerFunc(webpath, "web.version")
+		}
+		if err != nil {
+			log.Fatal().Msgf("No version found for product %s", product)
+		}
+		version = v
+	}
+	return version, product
+}
+
+func ScannerFunc(path string, search string) (string, error) {
+	var output string
+	f, err := os.Open(path)
+	if err != nil {
+		log.Fatal().Msg("Couldn't open file")
+	}
+	defer f.Close()
+	scanner := bufio.NewScanner(f)
+	for scanner.Scan() {
+		if strings.Contains(scanner.Text(), search) {
+			splits := strings.Split(scanner.Text(), "= ")
+			output = splits[1]
+			return output, nil
+		}
+	}
+	return "", fmt.Errorf("String not found!")
```

Comment:
why return an empty string and not just `nil`?

Reply:
Because ScannerFunc returns: string, error. It wont accept a nil